nuxeo-platform-spreadsheet-web/src/main/js/app/nuxeo/rest/schemas.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and contributors.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Lesser General Public License
+ * (LGPL) version 2.1 which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/lgpl-2.1.html
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+import {Request} from './request';
+
+const PATH = '/config/schemas/';
+
+/**
+ * REST Schemas
+ */
+class Schemas extends Request {
+
+  constructor(connection) {
+    super(connection, PATH);
+    this.data = {};
+  }
+
+  fetchFieldsBySchema(schema) {
+    return new Promise((resolve, reject) => {
+      this.conn.request(`${PATH}/${schema}?fetch.schema=fields`).repositoryName(undefined).get((error, data) => {
```

Comment:
I'd rather rely on `return this. execute('get', path).then(...)` or introduce a new method like `fetch` which returned a promise already